opaque_keys/edx/keys.py

```diff
@@ -61,6 +61,13 @@ def make_asset_key(self, asset_type, path):  # pragma: no cover
         """
         raise NotImplementedError()
 
+    @abstractmethod
+    def make_course_key_v2(self):  # pragma: no cover
```

Comment:
@zubair-arbi: You should have some a method to go from `CourseKeyV2` to `CourseKey` and back again, but only one of each. I would suggest using regular methods, rather than class methods, in general. So, probably `CourseKeyV2.make_course_run_key`, and `CourseKey.make_course_key_v2`.